sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java

```diff
@@ -238,11 +238,24 @@ public Node visitCreateTable(SqlBaseParser.CreateTableContext context) {
         return new CreateTable(
             (Table) visit(context.table()),
             visitCollection(context.tableElement(), TableElement.class),
-            visitCollection(context.crateTableOption(), CrateTableOption.class),
+            processTableOptions(context.partitionedByOrClusteredInto()),
             extractGenericProperties(context.withProperties()),
             notExists);
     }
 
+    private List<CrateTableOption> processTableOptions(@Nullable  SqlBaseParser.PartitionedByOrClusteredIntoContext ctx) {
+        ArrayList<CrateTableOption> tableOptions = new ArrayList<>(2);
+        if (ctx != null) {
```

Comment:
minor nitpick: Could return en empty list if the ctx is null as a shortcut and avoid the arraylist creation.